tests/TestCase/Shell/Task/TemplateTaskTest.php

```diff
@@ -446,6 +446,25 @@ public function testBakeIndex()
     }
 
     /**
+     * test bake template with index limit overwrite
+     *
+     * @return void
+     */
+    public function testBakeIndexWithIndexLimit()
+    {
+        $this->Task->controllerName = 'TemplateTaskComments';
+        $this->Task->modelName = 'TemplateTaskComments';
+        $this->Task->controllerClass = __NAMESPACE__ . '\TemplateTaskCommentsController';
+        $this->Task->params['indexColumns'] = 3;
```

Comment:
Yes I completely agree with you.
I believe it's a just a mistake that was not caught because the tests pass since it's written the same both in the test and in the code. 
It should be ``index-columns`` and it will work as expected.